### PR TITLE
Fix replay strings parsing

### PR DESCRIPTION
**Pull request by Kazbek**
Fix encoding problem described in #28 . 
It's really cause https://github.com/FAForever/downlords-faf-client/issues/1395 .
I`m tested correct displaying (see screenshot).
faf-scrn

### CODE REVIEW

File: faf-commons-data/src/main/java/com/faforever/commons/replay/ReplayDataParser.java [hunk 1]
```diff
@@ -51,12 +53,12 @@ public ReplayDataParser(Path path) {
   }
 
   private String readString(LittleEndianDataInputStream dataStream) throws IOException {
-    StringBuilder stringBuilder = new StringBuilder();
-    char character;
-    while ((character = (char) dataStream.readByte()) != 0) {
```
Comment (norraxx) [inline faf-commons-data/src/main/java/com/faforever/commons/replay/ReplayDataParser.java:56]: Do i understand right, that here we read each character as single byte? So there is not possible to have utf multi-byte characters?

File: faf-commons-data/src/main/java/com/faforever/commons/replay/ReplayDataParser.java [hunk 2]
```diff
@@ -51,12 +53,12 @@ public ReplayDataParser(Path path) {
...
-      stringBuilder.append(character);
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    byte tempByte;
+    while ((tempByte =  dataStream.readByte()) != 0) {
```
Comment (norraxx) [inline faf-commons-data/src/main/java/com/faforever/commons/replay/ReplayDataParser.java:60]: I saw this https://github.com/google/guava/blob/master/guava/src/com/google/common/io/LittleEndianDataInputStream.java#L176

Is it possible to use in our case?

Comment (norraxx) [reply faf-commons-data/src/main/java/com/faforever/commons/replay/ReplayDataParser.java:60]: I don't know the multi-byte utf, so i'm not sure, if we can read until value in not 0.

**Comment by norraxx** [comment]
I'll try to find at weekend some cases for testing. As @Askaholic said, it should be mojibake

**Comment by norraxx** [comment]
@Askaholic do you remember that encoding cases, that are not in utf? Maybe you have some interesting  examples?

**Comment by Kazbek** [comment]
@norraxx I read in wikipedia that multi-byte utf-8 symbol is never contains 0x0 octets (octets 2-4 starts with 10), so we can use it (how it was before) as delimeter. 
Also mentioned here https://stackoverflow.com/questions/6907297/can-utf-8-contain-zero-byte

File: faf-commons-data/src/main/java/com/faforever/commons/replay/ReplayDataParser.java (hunk 1 above)
Comment (Kazbek) [reply faf-commons-data/src/main/java/com/faforever/commons/replay/ReplayDataParser.java:56]: Yes, as example russian symbols looks like 110XXXXX 10XXXXXX so they dont dispay correctly in replay details window.

File: faf-commons-data/src/main/java/com/faforever/commons/replay/ReplayDataParser.java (hunk 2 above)
Comment (norraxx) [reply faf-commons-data/src/main/java/com/faforever/commons/replay/ReplayDataParser.java:60]: Anyway, can we read use `readUTF` for our case? There should be some sort of implementation: https://github.com/openjdk-mirror/jdk7u-jdk/blob/master/src/share/classes/java/io/DataInputStream.java#L588
I think, we won't need to have loop afterwards.

Comment (Kazbek) [reply faf-commons-data/src/main/java/com/faforever/commons/replay/ReplayDataParser.java:60]: I`m tried to use this method but all I have is 

> "Caused by: java.io.UTFDataFormatException: malformed input around byte 117"

As I understand readUTF read whole stream (into one String) but in this case we need to read stream part by part (one stream is passed many times in this method and loop checking that data is still available).

Comment (norraxx) [reply faf-commons-data/src/main/java/com/faforever/commons/replay/ReplayDataParser.java:60]: Ok, for me that one what you've done makes now sense

Comment (norraxx) [reply faf-commons-data/src/main/java/com/faforever/commons/replay/ReplayDataParser.java:60]: Good job ;-)

**Comment by Brutus5000** [comment]
The unit test fails on the old implementation populating the resulting string with weird characters.

**Review by Brutus5000 — APPROVED** [review]

**Pull request merged**